### ISSUE TITLE
Feature: Easily customize the internal node like ParagraphNode

### ISSUE THREAD

**Issue by aquarius-wing** [facebook/lexical#5867]
The ParagraphNode is hard core in lexical, and called in selection related function directly.

If we want to customize the ParagraphNode just like change the HTML it create, it will be difficult.

The key problem is, if we create a MParagraphNode to replace the type 'paragraph',

when we call the function inside $createParagraph, it comes the error:

```
Error: Create node: Type paragraph in node k does not match registered node ParagraphNode with the same type
```

**Comment by aquarius-wing** [comment]
I found it was solved, here is the #1262 

# Here is the summarized solution provided below:

Here is the [example](https://codesandbox.io/p/sandbox/ecstatic-maxwell-kw5utu?file=%2Fsrc%2Fnodes%2FCustomParagraphNode.js%3A14%2C18-14%2C28&from-embed= )


1. Define a class CustomParagraphNode 
  1. Notice here must be a new type
  2. Notice here must inherit the class you want to replace such as ParagraphNode
```javascript
import { ParagraphNode } from "lexical";

export class CustomParagraphNode extends ParagraphNode {
  static getType() {
    return "custom-paragraph";
  }

  static clone(node) {
    return new CustomParagraphNode(node.__key);
  }

  createDOM(config) {
    const dom = super.createDOM(config);
    dom.style = "background: green";
    return dom;
  }
}
```

2. In Editor.js, do not register the node which define in lexical such as ParagraphNode, the only thing you do is register new one such as CustomParagraphNode, and so on add a function help lexical to replace ParagraphNode to your CustomParagraphNode.

```javascript
    CustomParagraphNode,
    {
      replace: ParagraphNode,
      with: (node) => {
        return new CustomParagraphNode();
      }
    }
```
3. At last, if you want to write a class which can be customized by external, such as ParagraphNode, you just add a function  $applyNodeReplacement

```javascript
export function $createParagraphNode(): ParagraphNode {
  return $applyNodeReplacement(new ParagraphNode());
}
```

Here is the official [example](https://github.com/meogic-tech/lexical-for-meogic/blob/6e4f072ea194e7d3e631e488ad1ca1be79add50e/packages/lexical/src/nodes/LexicalParagraphNode.ts#L161-L160)

**Comment by vpzomtrrfrt** [comment]
But what if I want it to still serialize with the original type?
